pkg/kv/kvserver/rangefeed_restarter.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package kvserver
+
+import (
+	"context"
+	"math"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+)
+
+var rangefeedRestartInterval = settings.RegisterDurationSetting(
+	settings.SystemOnly,
+	"kv.rangefeed.restart_interval",
+	"minimum interval between rangefeed restarts to conform to new settings",
+	50*time.Millisecond,
+	settings.WithVisibility(settings.Reserved),
+)
+
+var rangefeedRestartAvailablePercent = settings.RegisterIntSetting(
+	settings.SystemOnly,
+	"kv.rangefeed.restart_quota_headroom",
```

Comment:
nit: let's carve out a namespace under `kv.rangefeed.restarter.*`.

The setting should probably also specify that it's catchup scans. Something like `max_catchup_fraction` or whatever.